packages/monaco/src/browser/monaco-outline-contribution.ts

```diff
@@ -83,26 +83,27 @@ export class MonacoOutlineContribution implements FrontendApplicationContributio
         if (editor) {
             const model = MonacoEditor.get(editor)!.getControl().getModel();
             this.toDisposeOnEditor.push(model.onDidChangeContent(() => this.updateOutline()));
+            this.toDisposeOnEditor.push(editor.editor.onSelectionChanged(selection => this.updateOutline(selection)));
```

Comment:
Refetching the whole outline on each cursor move looks expensive. Can we rely on last fetched outline?

Reply:
You're right. Sure. Thanks!

Reply:
I have added a super naive cache, which stores the most recently resolved outline "model" unless the model version (or the URI) changes.

So once it is loaded, and only the selection changes in the same editor, it does not fetch new data from the LS.

Reply:
At what point is it invalidated? I think the outline is always constructed only for one editor, probably something like following should work:
```ts
this.toDisposeOnEditor.push(Disposable.create(() => this.cachedRoots.clean()));
```
But then a map is not necessary, it can be:
```ts
protected cachedRoots: MonacoOutlineSymbolInformationNode[] | undefined;
...
this.toDisposeOnEditor.push(Disposable.create(() => this.cachedRoots = undefined));
```

Reply:
Cool, thanks for explaining. I thought we share one outline per application.